.github/workflows/conda-deps.yml

```diff
@@ -10,4 +10,5 @@ dependencies:
   - idyntree
   - spdlog
   - catch2>=3.3.0
+  - manif
```

Comment:
Is this a duplicate of `ci_env.yml`?

Reply:
It is not clear to me why this is necessary (see https://github.com/ami-iit/biomechanical-analysis-framework/pull/15#discussion_r1423113600). However, if we want to do so, perhaps we want to use more descriptive names? Something use the same directory for both files, and have names like `ci_env_linux.yml`  and `ci_env_macos_win.yml` ?